tests/Doctrine/Tests/ORM/Functional/Ticket/GH5742Test.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH5742Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            GH5742Person::class,
+            GH5742Toothbrush::class,
+            GH5742ToothpasteBrand::class
+        );
+    }
+
+    public function testUpdateOneToOneToNewEntityBeforePreviousEntityCanBeRemoved(): void
+    {
+        $this->expectNotToPerformAssertions();
```

Comment:
Maybe it should perform assertions? Right now, a no-op entity manager would pass the test. We could clear the EM after flush, select our person and assert that the right toothbrush is attached. Also, selecting the old toothbrush should yield an empty result.

Reply:
I'll add assertions, that makes it better

Reply:
done